addons/default/visiosoft/profile-module/src/Http/Controller/CacheController.php

```diff
@@ -1,15 +1,14 @@
 <?php namespace Visiosoft\ProfileModule\Http\Controller;
 
 use Anomaly\Streams\Platform\Http\Controller\PublicController;
-use Anomaly\UsersModule\User\Contract\UserRepositoryInterface;
-use Illuminate\Http\Request;
 
 class CacheController extends PublicController
 {
-    public function getUserInfo(Request $request, UserRepositoryInterface $userRepository)
+    public function getUserInfo()
     {
-        $user = $userRepository->find($request->userId);
+        $user = auth()->user();
+        $response = $user ? $user->first_name . ' ' . $user->last_name : $user;
 
-        return ['userName' => $user->first_name . ' ' . $user->last_name];
+        return ['userName' => $response];
```

Comment:
actually it's needed to return userid, and if user has store @Diatrex  Then you can render menu

Reply:
actually we don't have cache controller. It's just service that returning user data back @Diatrex

Reply:
I removed the addBlock function from the twig file. now we are getting the addblock from the controller